src/cropduster.js

```diff
@@ -265,48 +294,51 @@ const CD = {
     var calledIndex = -1;
     var msg = "getImages";
 
-    for(var i = 0; i < urls.length; i++) {
-      (function(url, i){
+    return new Promise(function(resolve, reject) {
```

Comment:
With the `getImagePromise` method, we could also rewrite `getImages` to leverage that directly and simplify quite a bit (this is doing a lot of manual bookkeeping!).